### PR TITLE
489 job chat centric candidate and admin messaging

### LINKED ISSUE #489: Job Chat centric candidate and admin messaging
As Talent Catalog
I would like to communicate with candidates and admin users in a more standard and scalable way
so that candidate and other user communication is simplified

- [x] #734
- [ ] #735
- [ ] #736
- [x] #737
- [ ] #738
- [x] #739

### PR DESCRIPTION
**Pull request by camerojo**

### CODE REVIEW

File: server/src/main/java/org/tctalent/server/service/db/JobChatService.java [hunk 1]
```diff
@@ -113,6 +115,26 @@ public interface JobChatService {
...
+     * @param ids The ids to be looked up. If an id does not correspond to any chat,
+     *            it is ignored and no chat is returned.
+     * @return Chats with ids matching the given ids.
+     */
+    List<JobChat> findByIds(@NonNull Collection<Long> ids);
+
+    /**
+     * Find chats which have posts where the date of the last post is greater than a given date.
+     * <p/>
+     * Note that this is in this service rather that JobChatService to avoid circular dependency.
```
Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/service/db/JobChatService.java]: Looks like a typo here as this is in the JobChatService

File: server/src/main/java/org/tctalent/server/service/db/impl/CandidateOpportunityServiceImpl.java [hunk 2]
```diff
@@ -552,4 +549,89 @@ private GoogleFileSystemFile uploadFile(String folderLink, String fileName,
...
+
+    /**
+     * Can be called directly by SystemAdminApi as often as needed without running into
+     * the SchedulerLock which is on {@link #scheduledNotifyOfChatsWithNewPosts()}
+     */
+    public void notifyOfChatsWithNewPosts() {
+
+        Map<Long, Set<JobChat>> userNotifications = new HashMap<>();
+
+        OffsetDateTime yesterday = OffsetDateTime.now().minusDays(30);
```
Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/service/db/impl/CandidateOpportunityServiceImpl.java]:
```suggestion
        OffsetDateTime yesterday = OffsetDateTime.now().minusDays(1);
```

**Review by sadatmalik — DISMISSED** [review]
Approved

File: server/src/main/java/org/tctalent/server/service/db/JobChatService.java (hunk 1 above)
Comment (camerojo) [reply server/src/main/java/org/tctalent/server/service/db/JobChatService.java]: Updated. Thanks!

**Review by sadatmalik — APPROVED** [review]
Approved

**Pull request merged**